rir/src/compiler/pir/bb.cpp

```diff
@@ -182,7 +182,11 @@ bool BB::before(Instruction* a, Instruction* b) const {
     }
     assert(false);
     return false;
-};
+}
+
+// TODO: More robust version. Should be based on real cfg. But on some analysis
+// it is expensive to compute it only for this behavior.
+bool BB::beforeInCfg(BB* otherBB) const { return this->id < otherBB->id; }
```

Comment:
we currently have no such guarantees currently, that does not work

Reply:
Empirically works. The commment just expose that, to make it sound we need to calculate the `cfg` for this pass. I guess that will increase compilation time significantly. Clearly it is easy to do. I just was not sure it is worth to pay for the overhead at this point. We probably should have a cfg for compiled function that is mutable and accessed from anywhere?

Reply:
> Empirically works

?? what does it even mean, e.g. in a loop

Reply:
It is implemented as a `<`. So the semantics it is intending to implement are `cfg.strictilyPreceeds(a,b)`

Reply:
well, the point is that this is currently only correct after the cleanup phase.

I am open to making this the default, since it might help us in many places. but not like this.

we could extract the renumbering from the cleanup phase and then have a renumber pass, that you can schedule before every pass where you want to rely on this property. and then some boolean on the function that tells you if the bb's are ordered. then we can clear it whenever we change sth and we can assert that it is set when we need it.